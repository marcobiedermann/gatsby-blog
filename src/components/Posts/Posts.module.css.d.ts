declare const styles: {
  readonly posts: string;
};
export = styles;
