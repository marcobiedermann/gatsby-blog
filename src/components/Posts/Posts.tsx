import React from 'react';
import PostPreview, { PostPreviewProps } from '../PostPreview/PostPreview';
import * as styles from './Posts.module.css';

export interface PostsProps {
  posts: PostPreviewProps[];
}

function Posts(props: PostsProps) {
  const { posts } = props;

  return (
    <ol className={styles.posts}>
      {posts.map((post) => {
        const { id } = post;

        return (
          <li key={id}>
            <PostPreview {...post} />
          </li>
        );
      })}
    </ol>
  );
}

export default Posts;
