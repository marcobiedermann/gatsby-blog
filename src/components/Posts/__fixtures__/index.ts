import { PostsProps } from '..';
import postPreview from '../../PostPreview/__fixtures__';

const posts: PostsProps = {
  posts: [postPreview],
};

export default posts;
