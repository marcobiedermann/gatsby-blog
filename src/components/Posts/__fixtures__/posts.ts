import { PostsProps } from '../Posts';
import postPreview from '../../PostPreview/__fixtures__/post-preview';

const posts: PostsProps = {
  posts: [postPreview],
};

export default posts;
