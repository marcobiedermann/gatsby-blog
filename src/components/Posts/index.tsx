import React, { FC } from 'react';
import PostPreview, { PostPreviewProps } from '../PostPreview';

export interface PostsProps {
  posts: PostPreviewProps[];
}

const Posts: FC<PostsProps> = (props) => {
  const { posts } = props;

  return (
    <ol>
      {posts.map((post) => {
        const { id } = post;

        return (
          <li key={id}>
            <PostPreview {...post} />
          </li>
        );
      })}
    </ol>
  );
};

export default Posts;
