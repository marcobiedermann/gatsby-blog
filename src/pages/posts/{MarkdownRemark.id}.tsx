import { graphql, PageProps } from 'gatsby';
import React, { FC } from 'react';
import Button from '../../components/Button';
import Layout from '../../components/Layout';

export interface DataProps {
  markdownRemark: {
    frontmatter: {
      title: string;
    };
    html: string;
  };
}

const PostPage: FC<PageProps<DataProps>> = (props) => {
  const {
    data: { markdownRemark },
  } = props;

  const { html } = markdownRemark;

  return (
    <Layout>
      <div dangerouslySetInnerHTML={{ __html: html }} />
      <p>
        <Button to="/">back</Button>
      </p>
    </Layout>
  );
};

export const query = graphql`
  query($id: String!) {
    markdownRemark(id: { eq: $id }) {
      frontmatter {
        title
      }
      html
    }
  }
`;

export default PostPage;
