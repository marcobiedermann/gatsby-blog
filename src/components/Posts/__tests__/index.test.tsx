import { render } from '@testing-library/react';
import React from 'react';
import Posts from '..';
import posts from '../__fixtures__';

describe('posts component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Posts {...posts} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
