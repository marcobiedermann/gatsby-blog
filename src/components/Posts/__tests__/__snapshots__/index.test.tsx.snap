// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts component should render correctly 1`] = `
<DocumentFragment>
  <ol>
    <li>
      <article>
        <h2>
          <a
            href="/blogfirst-post"
          >
            First Post
          </a>
        </h2>
        <div>
          <time
            datetime="2000-01-01"
          >
            01 Jan. 2000
          </time>
           â€¢ 1 mins.
        </div>
        <p>
          Lorem ipsum dolor sit amet.
        </p>
      </article>
    </li>
  </ol>
</DocumentFragment>
`;
